### PR TITLE
Add grid

### LINKED ISSUE #9: Create basic component to generate grid system

### PR DESCRIPTION
**Pull request by oskar-kupski-elpassion**
Closes #9

Remember to run `npm install` (scss and bootstrap 4 grid were added). You can configure grid for your needs in `src/components/grid/_grid.scss` file: breakpoints, max-widths, columns and gutter. Class names are taken directly from bootstrap framework

Also please keep in mind that I am newbie to Vue, so if anything is incorrect let me know

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,10 @@
     "build-storybook": "build-storybook"
   },
   "dependencies": {
+    "bootstrap-4-grid": "^3.1.1",
```
Comment (lubien) [inline package.json]: From https://github.com/vuejs-norte/puto-framework/issues/9#issuecomment-538069197

> this is a HTML-only framework 

No CSS allowed :)

> just put classes on divs

You did that already!

File: src/App.vue [hunk 2]
```diff
@@ -2,15 +2,75 @@
   <div id="app">
```
Comment (lubien) [inline src/App.vue]: Please don't do **anything** in this file

You need to create examples on `stories/` (look how other people did some)

File: src/components/grid/PCol.vue [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<template>
+  <div v-bind:class="[`col-${sizeXs}`,`col-sm-${sizeSm}`,`col-md-${sizeMd}`,`col-lg-${sizeLg}`,`col-xl-${sizeXl}`]">
```
Comment (lubien) [inline src/components/grid/PCol.vue]: `v-bind:class`

`:class`

Comment (lubien) [reply src/components/grid/PCol.vue]: Also please make a computed property called `classNames`.

If there is no `sizeXs` **don't** add `col-xs-undefined`.

File: src/components/grid/PCol.vue [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      sizeXs: Number,
+      sizeSm: Number,
+      sizeMd: Number,
+      sizeLg: Number,
+      sizeXl: Number
+    }
+  }
+</script>
+
+<style scoped lang="scss">
```
Comment (lubien) [inline src/components/grid/PCol.vue]: No CSS

File: src/components/grid/PCol.vue (hunk 3 above)
Comment (lubien) [reply src/components/grid/PCol.vue]: All class names should start with `puto-`

File: src/components/grid/PContainer.vue [hunk 5]
```diff
@@ -0,0 +1,30 @@
+<template>
+  <div v-bind:class="[fluid ? 'container-fluid' : 'container']">
```
Comment (lubien) [inline src/components/grid/PContainer.vue]: `class`

All class names should start with `puto-`

File: src/components/grid/PContainer.vue [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    <slot></slot>
+    <div class="row">
```
Comment (lubien) [inline src/components/grid/PContainer.vue]: ?

File: src/components/grid/PContainer.vue [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+<script>
+  export default {
+    name: 'PContainer',
+    props: {
+      fluid: Boolean
+    }
+  }
+</script>
+
+<style scoped lang="scss">
```
Comment (lubien) [inline src/components/grid/PContainer.vue]: No CSS

File: src/components/grid/PRow.vue [hunk 8]
```diff
@@ -0,0 +1,19 @@
+<template>
+  <div class="row">
```
Comment (lubien) [inline src/components/grid/PRow.vue]: All class names should start with `puto-`

File: src/components/grid/PRow.vue [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  </div>
+</template>
+
+<script>
+  export default {
+    name: "PRow"
+  }
+</script>
+
+<style scoped lang="scss">
```
Comment (lubien) [inline src/components/grid/PRow.vue]: No CSS

File: src/components/grid/_grid.scss [hunk 10]
```diff
@@ -0,0 +1,30 @@
+$grid-breakpoints: (
```
Comment (lubien) [inline src/components/grid/_grid.scss]: No CSS

**Review by lubien — CHANGES_REQUESTED** [review]
Don't modify our package{-lock}.json files :)

**Comment by lubien** [comment]
> Also please keep in mind that I am newbie to Vue, so if anything is incorrect let me know

Sure, let's do it together

File: package.json (hunk 1 above)
Comment (oskar-kupski-elpassion) [reply package.json]: So I should add classess to divs, but how should I style the divs without using css? Also where should I store information about media breakpoints? Should it be in css or dynamiclly calculated via js ?

**Comment by oskar-kupski-elpassion** [comment]
@lubien I've done some refactoring, please let me know it it works you better

File: src/components/grid/PCol.vue [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+  name: 'PCol',
+  props: {
+    sizeXs: Number,
+    sizeSm: Number,
+    sizeMd: Number,
+    sizeLg: Number,
+    sizeXl: Number
+  },
+  computed: {
+    className() {
```
Comment (lubien) [inline src/components/grid/PCol.vue:18]: Update to be like https://github.com/vuejs-norte/puto-framework/blob/master/src/components/PButton.vue#L40-L46

File: stories/PGrid.stories.js [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+import Render from './Render'
+
+export default {
+  title: 'PGrid'
+}
+
+export const basicUsage = () =>
+  Render({
+    components: { PContainer, PRow, PCol },
+    template: `<PContainer>
```
Comment (lubien) [inline stories/PGrid.stories.js:13]: Let's indent like this:

```js
    template: `<PContainer>
      <PRow>
        <PCol :size-sm="10" :size-lg="12">COLUMN</PCol>
        <PCol :size-md="10" :size-xl="12">COLUMN</PCol>
        <PCol :size-xs="10" :size-lg="12">COLUMN</PCol>
      </PRow>
      <PRow>
        <PCol :size-xs="10" :size-lg="12">COLUMN</PCol>
      </PRow>
    </PContainer>`
  })

```

File: stories/PGrid.stories.js [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+                 <PRow>
+                   <PCol :size-xs="10" :size-lg="12">COLUMN</PCol>
+                 </PRow>
+               </PContainer>`
+  })
+
+export const fluidContainer = () =>
+  Render({
+    components: { PContainer, PRow, PCol },
+    template: `<PContainer fluid>
```
Comment (lubien) [inline stories/PGrid.stories.js:28]: Same indent

File: package.json (hunk 1 above)
Comment (lubien) [reply package.json]: The styles are out of the scope of this framework. It's meant to be a HTML-only POC framework that you would implement stuff like that later.

We really need that info on our repo (there's an issue for this)

**Review by lubien — CHANGES_REQUESTED** [review]
Just one last review

**Comment by oskar-kupski-elpassion** [comment]
done and pushed :)

**Review by lubien — APPROVED** [review]

**Comment by lubien** [comment]
Thank you :3

**Pull request merged**